src/rpc/blockchain.cpp

```diff
@@ -346,14 +340,10 @@ static RPCHelpMan waitforblockheight()
                     HelpExampleCli("waitforblockheight", "100 1000")
             + HelpExampleRpc("waitforblockheight", "100, 1000")
                 },
-        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+        [](const RPCContext& ctx) -> UniValue
 {
-    int timeout = 0;
-
-    int height = request.params[0].get_int();
-
-    if (!request.params[1].isNull())
-        timeout = request.params[1].get_int();
+    int height = ctx.param<int>(0);
```

Comment:
This is a behavior change, as `height` is not optional; it would throw before, and will silently use height 0 now.

Reply:
> and will silently use height 0 now.

How? `param<int>(index)` results in the same code as before.

Reply:
I thought the 0 was a default value, not the index. You're right.